poetry/installation/chooser.py

```diff
@@ -70,7 +70,10 @@ def choose_for(self, package: Package) -> Link:
             links.append(link)
 
         if not links:
-            raise RuntimeError(f"Unable to find installation candidates for {package}")
+            raise RuntimeError(
+                f"Unable to find installation candidates for {package} "
+                f"matching digests: {[p['hash'] for p in package.files]}"
+            )
```

Comment:
Wouldn't that be misleading since there might be other reasons why no links could be found, right ?

Reply:
I'm not sure I follow; there are a bunch of criteria for a failing match, this is now simply showing more of them so users can work out what's gone wrong.  But I'm open to suggestions for a better error message if this one is no good.

Reply:
I force-pushed a new iteration of this change that throws a separate exception for this; that should reduce the ambiguity.